core/src/main/java/org/apache/iceberg/CachingCatalog.java

```diff
@@ -30,23 +35,98 @@
 import org.apache.iceberg.catalog.TableIdentifier;
 import org.apache.iceberg.exceptions.AlreadyExistsException;
 import org.apache.iceberg.relocated.com.google.common.collect.ImmutableList;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
+/**
+ * Class that wraps an Iceberg Catalog to cache table's.
+ *
+ * When `expirationIntervalMillis` is non-zero, the cache will expire tables after
+ * the given interval provided they have not been accessed via the catalog in that time.
+ * Each lookup of the table via the catalog will restart the expiration time.
+ *
+ * If the duration is zero, the cache will retain entries forever, or until
+ * an action is taken on the cache to refresh the cache's table entry.
+ */
 public class CachingCatalog implements Catalog {
+
+  private static final Logger LOG = LoggerFactory.getLogger(CachingCatalog.class);
+  private static final RemovalListener<TableIdentifier, Table> identLoggingRemovalListener =
+      (key, value, cause) -> LOG.debug("{} was evicted from the TableCache with {} cause", key, cause);
```

Comment:
We don't need this, right?

If you want to keep it, let's change it to be a bit shorter: `Evicted {} from table cache ({})`. Remember that the people getting these messages probably don't know any variable or class names, so things like `TableCache` should be formatted in plain English, "table cache".

Reply:
Fixed